vm/actor/src/builtin/miner/deadline_state.rs

```diff
@@ -0,0 +1,998 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::{
+    BitFieldQueue, ExpirationSet, Partition, PartitionSectorMap, PoStPartition, PowerPair,
+    QuantSpec, SectorOnChainInfo, Sectors, TerminationResult, WPOST_PERIOD_DEADLINES,
+};
+use crate::{actor_error, ActorError, ExitCode, TokenAmount};
+use bitfield::BitField;
+use cid::{multihash::Blake2b256, Cid};
+use clock::ChainEpoch;
+use encoding::tuple::*;
+use fil_types::SectorSize;
+use ipld_amt::Amt;
+use ipld_blockstore::BlockStore;
+use num_traits::Zero;
+use std::{cmp, collections::HashMap, collections::HashSet, error::Error as StdError};
+
+/// Deadlines contains Deadline objects, describing the sectors due at the given
+/// deadline and their state (faulty, terminated, recovering, etc.).
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct Deadlines {
+    // Note: we could inline part of the deadline struct (e.g., active/assigned sectors)
+    // to make new sector assignment cheaper. At the moment, assigning a sector requires
+    // loading all deadlines to figure out where best to assign new sectors.
+    pub due: Vec<Cid>, // []Deadline
```

Comment:
I think I tried it but ran into problems because `WPOST_PERIOD_DEADLINES` is greater than 32 😛 So some traits weren't implemented for it (I don't remember which one was the issue). I'll add a TODO to change this to an array later, let me know if you think that's sensible

Reply:
yeah lets just wait, that's fine. Hopefully the change away from that >32 trait gets removed at some point soon